packages/edit-widgets/src/store/actions.js

```diff
@@ -108,10 +108,15 @@ export function* saveWidgetArea( widgetAreaId ) {
 		( { sidebar } ) => sidebar === widgetAreaId
 	);
 
-	// Remove all duplicate reference widget instances
+	// Remove all duplicate reference widget instances for legacy widgets.
+	// Why? We filter out the widgets with duplicate IDs to prevent adding more than one instance of a widget
+	// implemented using a function. WordPress doesn't support having more than one instance of these, if you try to
+	// save multiple instances of these in different sidebars you will run into undefined behaviors.
 	const usedReferenceWidgets = [];
-	const widgetsBlocks = post.blocks.filter( ( { attributes: { id } } ) => {
-		if ( id ) {
+	const widgetsBlocks = post.blocks.filter( ( block ) => {
+		const { id } = block.attributes;
+
+		if ( block.name === 'core/legacy-widget' && block.attributes?.id ) {
```

Comment:
This seems like a typo :)?

```suggestion
		if ( block.name === 'core/legacy-widget' && id ) {
```

Reply:
yeah, thanks!